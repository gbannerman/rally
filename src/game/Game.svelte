<script lang="ts">
  import ScoreControls from "./ScoreControls.svelte";
  import Scores from "./Scores.svelte";
  import Timer from "../timer/Timer.svelte";

  let scores = [0, 0];

  const resetScores = () => {
    scores = [0, 0];
  };

  const addPoint = (index: number) => {
    scores[index] = scores[index] + 1;
  };
</script>

<Timer />
<Scores {scores} onScoreClick={addPoint} />
<ScoreControls onResetClick={resetScores} />

<style>
</style>
