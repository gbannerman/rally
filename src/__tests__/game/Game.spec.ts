import { render, fireEvent } from "@testing-library/svelte";

import Game from "../../game/Game.svelte";

describe("Game component", () => {
  it("renders correctly", () => {
    const { container } = render(Game);

    expect(container).toMatchSnapshot();
  });

  it("starts scores at zero", () => {
    const { getAllByRole } = render(Game);

    const buttons = getAllByRole("button", { name: "0" });

    expect(buttons).toHaveLength(2);
  });

  it("clicking on player 1 increases score by point", async () => {
    const { getAllByRole } = render(Game);

    const [player1Button, player2Button] = getAllByRole("button", {
      name: "0",
    });

    await fireEvent.click(player1Button);

    expect(player1Button).toHaveTextContent("1");
    expect(player2Button).toHaveTextContent("0");
  });

  it("clicking on player 2 increases score by point", async () => {
    const { getAllByRole } = render(Game);

    const [player1Button, player2Button] = getAllByRole("button", {
      name: "0",
    });

    await fireEvent.click(player2Button);

    expect(player1Button).toHaveTextContent("0");
    expect(player2Button).toHaveTextContent("1");
  });

  it("clicking reset score button resets both scores to zero", async () => {
    const { getAllByRole, getByRole } = render(Game);

    const [player1Button, player2Button] = getAllByRole("button", {
      name: "0",
    });
    const resetScoresButton = getByRole("button", { name: "Reset score" });

    await fireEvent.click(player1Button);

    await fireEvent.click(player2Button);
    await fireEvent.click(player2Button);

    await fireEvent.click(resetScoresButton);

    expect(player1Button).toHaveTextContent("0");
    expect(player2Button).toHaveTextContent("0");
  });
});
