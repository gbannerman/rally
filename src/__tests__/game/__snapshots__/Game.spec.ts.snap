// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders correctly 1`] = `
<body>
  <div>
    <div
      class="Timer svelte-u8xogd"
    >
      <div
        aria-label="Timer"
        class="TimerInput svelte-jd5xj"
      >
        <div
          class="timer placeholder svelte-jd5xj"
        >
          00:00
        </div>
         
        <form>
          <input
            class="timer input svelte-jd5xj"
            inputmode="numeric"
            maxlength="5"
            style="width: 10px; margin-right: 2px; padding-left: 0px; right: 0px;"
            type="text"
          />
        </form>
      </div>
       
      <button
        class="icon start svelte-znyfjv"
      >
        <svg
          class="feather feather-play "
          fill="none"
          height="100%"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="5 3 19 12 5 21 5 3"
          />
        </svg>
      </button>
    </div>
     
    <div
      class="Scores svelte-c6f6eo"
    >
      <button
        class="score player1 svelte-c6f6eo"
      >
        <div
          class="svelte-c6f6eo"
        >
          0
        </div>
      </button>
       
      <button
        class="score player2 svelte-c6f6eo"
      >
        <div
          class="svelte-c6f6eo"
        >
          0
        </div>
      </button>
    </div>
     
    <div
      class="PointDifferenceIndicator svelte-16qa96h"
    >
       
    </div>
     
    <div
      class="ScoreControls svelte-1p4m5fg"
    >
      <button
        class="svelte-1p4m5fg"
      >
        Reset score
      </button>
    </div>
  </div>
</body>
`;
